zvariant/src/display.rs

```diff
@@ -0,0 +1,601 @@
+/// Implemented based on https://gitlab.gnome.org/GNOME/glib/-/blob/e1d47f0b0d0893ac9171e24cc7bf635495376546/glib/gvariant.c#L2213
+use std::{
+    ffi::CStr,
+    fmt::{Display, Write},
+};
+
+use crate::{Array, Dict, Structure, Value};
```

Comment:
I'd rather keep the `Display` impls of each type in their containing file. That's not just a general convention but also consistent with other impls in our crate and repo.

Moroever, I'd ask you to add impl for each type in a separate commit so it'd be easier to add the changes to separate commits if each impl is added to the appropriate files.

Reply:
> Moroever, I'd ask you to add impl for each type in a separate commit so it'd be easier to add the changes to separate commits if each impl is added to the appropriate files.

I think that would not be easy since the Display impl of `Value` depends on the Display impl of `Maybe` etc. and vice versa, i.e., `value_display_fmt` calls on `maybe_display_fmt` etc. and `maybe_display_fmt` calls on `value_display_fmt`.

Reply:
I'm also a bit confused about the best way to organize these `*_display_fmt` functions. Should I store them all in `value.rs` and just call them on `maybe.rs` etc. or store them in their respective files (e.g., `maybe.rs`) and just call them in `value.rs`?

Reply:
> > I think that would not be easy since the Display impl of Value depends on the Display impl of Maybe etc. and vice versa, i.e., value_display_fmt calls on maybe_display_fmt etc., and maybe_display_fmt etc. calls on value_display_fmt.
>
> Is that a problem?

I meant it with splitting the commit into for each file

Reply:
> I meant it with splitting the commit into for each file

Ah, sorry I didn't get it. I see the problem. Yeah, that's tough then. I guess this is an atomic commit since all modules need to implement `Display` at the same time to be able to build even. Forgot what I said about splitting commits. Sorry, should have thought about the mutual dependencies.

Reply:
No worries